src/class-big-pit.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * Big_Pit class file
+ *
+ * @package wp-big-pit
+ */
+
+// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery
+// phpcs:disable WordPress.DB.DirectDatabaseQuery.NoCaching
+
+namespace Alley\WP;
+
+use Alley\WP\Types\Feature;
+
+/**
+ * The Big Pit.
+ */
+final class Big_Pit implements Feature {
+	/**
+	 * Singleton.
+	 *
+	 * @var self
+	 */
+	private static ?self $instance = null;
+
+	/**
+	 * Whether the database is ready to accept queries.
+	 *
+	 * @var bool
+	 */
+	private bool $ready = false;
+
+	/**
+	 * Instance.
+	 *
+	 * @return self
+	 */
+	public static function instance(): self {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+			self::$instance->boot();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Boot the feature.
+	 */
+	public function boot(): void {
+		global $wpdb;
+
+		$wpdb->big_pit = $wpdb->base_prefix . 'big_pit';
+
+		if ( defined( 'WP_INSTALLING' ) ) {
+			return;
+		}
+
+		$this->upsert();
+		$this->ready = true;
+	}
+
+	/**
+	 * Get a value.
+	 *
+	 * @param string $key   Item key.
+	 * @param string $group Item group.
+	 * @return mixed|null
+	 */
+	public function get( string $key, string $group ): mixed {
+		global $wpdb;
+
+		if ( ! $this->ready ) {
+			return null;
+		}
+
+		$value = $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT item_value FROM {$wpdb->big_pit} WHERE item_group = %s AND item_key = %s LIMIT 1",
+				$group,
+				$key
+			),
+		);
+
+		if ( is_string( $value ) ) {
+			$value = maybe_unserialize( $value );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Set a value.
+	 *
+	 * @param string $key   Item key.
+	 * @param mixed  $value Item value.
+	 * @param string $group Item group.
+	 */
+	public function set( string $key, mixed $value, string $group ): void {
```

Comment:
Might it be beneficial to return `bool` on the `set` and `delete` methods, to mirror the corresponding object cache methods?

Reply:
this got me thinking that maybe this should do more than mirror the object cache! expanded on that in another comment

Reply:
I've recently become somewhat skeptical of these return values. In addition to the possibility of ambiguity (does `false` mean the value wasn't deleted or that there was nothing to delete? would you return `true` even if no deletions took place?), have you found a use for them?